src/lib/support/jsontlv/README.md

```diff
@@ -0,0 +1,106 @@
+information, the Json key of each element should contain the TLV element tag and
+type information.
+
+The Json key schema is as follows:
+
+```
+{ [field_name:]field_id:element_type[-sub_element_type] }
+```
+
+Specific rules:
+
+-   'element_type' MUST map to the TLV Type as specified in the table below.
+-   For 'element_type' of type "ARRAY", the sub_element_type MUST always be
+    present and be equal to the 'element_type' of the elements in the array.
+-   Array elements MUST have the same 'element_type'.
+-   Array elements MUST have anonymous TLV tags.
+-   'field_name' is always an optional field and can be provided if the semantic
+    name for the field is available.
+-   In the case of an empty array, 'sub_element_type' should be "?" (unknown
+    element type).
+-   The unknown element type “?” MUST never occur outside of "ARRAY-?".
+-   'field_id' is equivalent to the Field ID for all cluster payloads (commands,
+    events, attributes).
+
+The table below summarizes all element types and their corresponding encoding in
+the Json key:
+
+| TLV Type                               | element_type           | JSON Type               |
+| -------------------------------------- | ---------------------- | ----------------------- |
+| Unsigned Integer ( < 2^32 )            | UINT                   | Number                  |
+| Unsigned Integer ( >= 2^32 )           | UINT                   | String                  |
+| Signed Integer ( >= -2^31 and < 2^31 ) | INT                    | Number                  |
+| Signed Integer ( < -2^31 or >= 2^31 )  | INT                    | String                  |
+| Boolean                                | BOOL                   | Boolean                 |
+| Float (32-bit)                         | FLOAT                  | Number                  |
+| Float/Double (64-bit)                  | DOUBLE                 | Number                  |
+| Octet String                           | BYTES                  | String (Base64 encoded) |
+| UTF-8 String                           | STRING                 | String                  |
+| Null Value                             | NULL                   | null                    |
+| Struct                                 | STRUCT                 | Dict                    |
+| Array                                  | ARRAY-sub_element_type | Array                   |
+| Unknown                                | ?                      | Empty Array             |
+
+Note that this is NOT a generalized JSON representation of TLV and does not
+support arbitrary TLV conversions. The main goal of this Json format is to
+support data model payloads for events/commands/attributes. Some of the
+limitations of this format are:
+
+-   TLV List types are not supported.
+-   TLV Array cannot contain another TLV Array.
+-   The top-level container MUST be an anonymous STRUCT.
+-   Infinity Float/Double values are not supported.
```

Comment:
We do have some tests that uses positive and negative infinity: https://github.com/project-chip/connectedhomeip/blob/d2b1ea8c838e638331483ff98d651d83ba5cf887/src/app/tests/suites/TestCluster.yaml#L2572

My understanding is that this is part of the TLV format per `src/common_protocol/tlv-format.adoc`

Reply:
Yes, they are supported by CHIP TLV implementation but these infinity values are not supported by the specific Json CPP library that is used in the SDK. This is the source of the limitation

Reply:
JSON does not support Infinity, sure.  You can encode it as a string like you are doing for similarly-unsupported ints that have > 2^53 absolute value.

But the point is, to support the actual claimed use case here as I understand it (represent data model payloads) you have to support Infinity.

Reply:
@bzbarsky-apple This is a good idea and I added support for the float/double infinity elements in the latest commit.